theseus/theseus_layer.py

```diff
@@ -9,22 +9,20 @@
 import torch.nn as nn
 from torch.autograd.function import once_differentiable
 
-from theseus.core import Variable
-from theseus.core.cost_function import AutoDiffCostFunction
+from theseus.core import AutoDiffCostFunction, Variable, Vectorize
 from theseus.optimizer import Optimizer, OptimizerInfo
 from theseus.optimizer.linear import LinearSolver
 from theseus.optimizer.nonlinear import BackwardMode, GaussNewton
 
 
 class TheseusLayer(nn.Module):
-    def __init__(
-        self,
-        optimizer: Optimizer,
-    ):
+    def __init__(self, optimizer: Optimizer, vectorize_cost_fns: bool = True):
```

Comment:
When we vectorize other things like cost weights or retraction (anything else?) would the interface need a separate bool for all of them or should we have a single `vectorize` for everything?

Reply:
I prefer the single `vectorize` for everything, seems cleaner. Any thoughts?

Reply:
Sounds fine to me, @fantaosha?

Reply:
The vectorization of the retraction map should be the default choice and always `True`. I'm not sure if the vectorization of weights and robust losses would raise any problems. In particular, many cost functions might share weights and robust losses, and this might be a problem for `AutoDiffCostFunction` using `functorch`.

Reply:
In addition, I didn't see a performance gain for the vectorization of weights and robust losses in the experimental branch.

Reply:
Renamed as `vectorize: bool = True`.